### PR TITLE
Boundary & misc fixes for MC

### PR DESCRIPTION
**Pull request by AstroBarker**
This PR adds a few fixes / enhancements for monte carlo

- More support for swarm user boundaries, needed for torus's BCs.
- Add the Ucon0 term for monte carlo absorption Ye update
- Bug fix in the monte_carlo boundary bit, add in `swarm->ResetCommunication();` to stop strange crash.

### CODE REVIEW

File: src/radiation/monte_carlo.cpp [hunk 1]
```diff
@@ -419,6 +419,9 @@ TaskStatus MonteCarloTransport(MeshBlock *pmb, MeshBlockData<Real> *rc,
               v(prho, k, j, i), v(itemp, k, j, i), v(iye, k, j, i), s, nu);
 
           Real dtau_abs = alphanu * dt; // c = 1 in code units
+          Real Ucon[4];
+          Real vel[3] = {v(ivlo, k, j, i), v(ivlo + 1, k, j, i), v(ivlo + 2, k, j, i)};
+          GetFourVelocity(vel, geom, CellLocation::Cent, k, j, i, Ucon);
```
Comment (Yurlungur) [inline src/radiation/monte_carlo.cpp:424]: Don't need full four-velocity so you can just compute the 0th component
```suggestion
          Real vel[3] = {v(ivlo, k, j, i), v(ivlo + 1, k, j, i), v(ivlo + 2, k, j, i)};
          Real W = GetLorentzFactor(v, geom, CellLocation::Cent, k, j, i);
          Real alpha = geom.Lapse(CellLocation::Cent, k, j, i);
          Real Ucon0 = robust::ratio(W, std::abs(alpha));
```

**Review by Yurlungur — APPROVED** [review]

File: src/radiation/monte_carlo.cpp (hunk 1 above)
Comment (AstroBarker) [reply src/radiation/monte_carlo.cpp:424]: Fair that's cleaner, will do

**Review by brryan — APPROVED** [review]
LGTM, thanks for these fixes @AstroBarker! I agree with @Yurlungur about just calculating u^0 but here is my approval now.

**Comment by Yurlungur** [comment]
@AstroBarker let me know when it's ready for merge. Probably can go in once tests pass.

**Comment by AstroBarker** [comment]
> @AstroBarker let me know when it's ready for merge. Probably can go in once tests pass.

I'm happy with it once tests pass.

**Pull request merged**